packages/teleport/src/DesktopSession/useDesktopSession.tsx

```diff
@@ -40,34 +42,90 @@ export default function useDesktopSession(ctx: Ctx) {
   // disconnected tracks whether the user intentionally disconnected the client
   const [disconnected, setDisconnected] = useState(false);
 
+  const [clipboardSharingPermitted, setClipboardSharingPermitted] =
+    useState(false);
+
   const { username, desktopName, clusterId } = useParams<UrlDesktopParams>();
   const [hostname, setHostname] = useState<string>('');
+
+  const isUsingChrome = navigator.userAgent.indexOf('Chrome') > -1;
+  const clipboardRWPermission = useClipboardReadWrite(
+    isUsingChrome && clipboardSharingPermitted
+  );
+  const [clipboard, setClipboard] = useState({
+    enabled: clipboardSharingPermitted,
+    permission: clipboardRWPermission,
```

Comment:
I could try that but it would require a lot of refactoring, and moving a lot of DesktopSession component level logic into the `useClipboard` hook. For example, right now if `isUsingChrome = false` and `clipboardSharingPermitted = true`, `useClipboardReadWrite` will give me `"unknown"`, but at the component level that's an error. Vs if `isUsingChrome = false` and `clipboardSharingPermitted = false`, I'll also get an `"unknown"` but at the component level that's not an error. 

Right now which of those is an error is being managed in the [`useEffect` below](https://github.com/gravitational/webapps/blob/d754b540a6d580445da76fff05cd7aee8d035b6d/packages/teleport/src/DesktopSession/useDesktopSession.tsx#L73-L108). It's simpler to understand and more maintainable if this hook to manage DesktopSession's state whereas the `useClipboard` hook just manages the browser clipboard permissions state.